### DISCUSSION TITLE
Avoiding the 'more' prompt with single-line messages on `cmdheight=0`

### DISCUSSION THREAD

**Discussion by LunarLambda** [neovim/neovim#30539] [Ideas]
Tried `cmdheight=0` again today after not touching it for a long time to see if it had gotten better. Saw that single-line messages (like `:e` without a filename) still cause the 'Press ENTER or type command' prompt to show.

So I'm curious what potential designs could look like for avoiding this. Two possibilities that come to mind:

Should the statusline have a 'message' field where it can insert the message text? This would require more integration, but would potentially be more flexible.

or

Should there be a sort of timeout value, where the message shows instead of the statusline, and then disappears after e.g. 1 or 2 seconds? Has some precedent with e.g. the `hlsearch` plugin, and I would personally prefer

or maybe the maintainers have other ideas in mind already?

**Comment by Shougo** [comment]
Hi, I am the original author for the feature.

Unfortunately, this is by design.

In Vim/neovim, the message string is displayed on the command line. Since set cmdheight=0 hides the command line, there is no place to display the message string. When there is not enough space to display the message string, Vim/neovim shows the 'more' prompt.

> Should the statusline have a 'message' field where it can insert the message text? This would require more integration, but would potentially be more flexible.

I don't think this is the solution because there are people who want to hide both the status line and the command line (and I am one of them).

> Should there be a sort of timeout value, where the message shows instead of the statusline, and then disappears after e.g. 1 or 2 seconds? Has some precedent with e.g. the hlsearch plugin, and I would personally prefer

Hm... This?

https://github.com/neovim/neovim/pull/1802

I think it should be printed by plugins.

Please see this.

https://github.com/neovim/neovim/issues/27811

**Comment by LunarLambda** [comment]
Wasn't yet aware of #27811, thank you. Guess I was retreading old ground a little

I think for me personally, fading messages in the statusline area (maybe using CursorMoved) + having excess messages shown as `+n messages`  would solve most of my pain points and wishes towards `cmdheight=0`. I've subscribed to the issue to follow the development

**Reply by Shougo** [reply]
> I think for me personally, fading messages in the statusline area (maybe using CursorMoved) + having excess messages shown as +n messages would solve most of my pain points and wishes towards cmdheight=0. I've subscribed to the issue to follow the development

It is possible.  But current messaging implementation is very complex.  It is hard to change...

**Comment by justinmk** [comment]
> Should the statusline have a 'message' field where it can insert the message text?

Yeah, that's basically the plan. https://github.com/neovim/neovim/issues/22478
